packages/core/src/behavior.ts

```diff
@@ -255,46 +258,46 @@ export function createMachineBehavior<TContext, TEvent extends EventObject>(
   let resolvedMachine: StateMachine<TContext, TEvent>;
 
   const behavior: Behavior<TEvent, State<TContext, TEvent>> = {
-    receiveSignal: (actorContext, signal) => {
-      resolvedMachine = isFunction(machine) ? machine() : machine;
-
-      if (signal === startSignal) {
-        service = interpret(resolvedMachine, {
-          ...options,
-          parent,
-          id: actorContext.name
-        });
-        service.onDone((doneEvent) => {
-          parent?.send(
-            toSCXMLEvent(doneEvent, {
-              origin: actorContext.self
-            })
-          );
-        });
-
-        if (options?.sync) {
-          subscription = service.subscribe((state) => {
+    receive: (state, event, actorContext) => {
+      if (isSignal(event)) {
+        resolvedMachine = isFunction(machine) ? machine() : machine;
+
+        if (event === startSignal) {
+          service = interpret(resolvedMachine, {
+            ...options,
+            parent,
+            id: actorContext.name
+          });
+          service.onDone((doneEvent) => {
             parent?.send(
-              toSCXMLEvent(
-                {
-                  type: actionTypes.update,
-                  state
-                },
-                { origin: actorContext.self }
-              )
+              toSCXMLEvent(doneEvent, {
+                origin: actorContext.self
+              })
             );
           });
+
+          if (options?.sync) {
```

Comment:
can we kill `sync` since we have `getSnapshot`?

Reply:
Yes, in a separate PR though because we want to be sure that we can support the pattern of receiving events via a subscription from an actor (which is basically what `sync` does, implicitly)